src/plugins/mer/merbuildengineoptionswidget.cpp

```diff
@@ -643,6 +644,7 @@ void MerBuildEngineOptionsWidget::onSrcFolderApplyButtonClicked(const QString &n
     buildEngine->virtualMachine()->lockDown(false, this, IgnoreAsynchronousReturn<bool>);
 
     if (ok) {
+        Core::DocumentManager::setProjectsDirectory(buildEngine->sharedSrcPath());
```

Comment:
Wouldn't it make sense to first check if current projectsDirectory already happens to be under workspace?

Reply:
Yup, done.